src/node_debug.cc

```diff
@@ -0,0 +1,101 @@
+#include "node_debug.h"
+
+#ifdef DEBUG
+#include "node_binding.h"
+
+#include "env-inl.h"
+#include "util.h"
+#include "v8-fast-api-calls.h"
+#include "v8.h"
+
+#include <string_view>
+#include <unordered_map>
+#endif  // DEBUG
+
+namespace node {
+namespace debug {
+
+#ifdef DEBUG
+using v8::Context;
+using v8::FastApiCallbackOptions;
+using v8::FunctionCallbackInfo;
+using v8::Local;
+using v8::Number;
+using v8::Object;
+using v8::Value;
+
+thread_local std::unordered_map<std::string_view, int> v8_fast_api_call_counts;
+
+void TrackV8FastApiCall(std::string_view key) {
+  v8_fast_api_call_counts[key]++;
```

Comment:
For the purpose of the tests, I think using a bool should be enough - an int would overflow if the process runs for long enough.

Also if we require the counter/toggle to be pre-declared in a macro list, we can just track them in thread_local bool instead of doing a runtime map lookup, and the overhead of simply setting a variable to false feels negligible enough to be enabled at runtime.

Reply:
> an int would overflow if the process runs for long enough

I know that but I don't think it matters. As an internal debug-only API, it is not meant to be useful outside of our unit tests and those would never run for long enough.

I thought a counter could be useful for complex cases where we want to call the same API multiple times (or it's called in a loop internally).

I'd say feel free to refactor / change in a future PR (or block this one). I spent enough energy on it myself.

Reply:
> I thought a counter could be useful for complex cases where we want to call the same API multiple times (or it's called in a loop internally).

Even if it's called multiple times, only V8 can really reliably check exactly how many times the fast API is run, as that's subject to the optimizing strategy - unless we use the intrinsics that are not supported for embedders' use cases. As an embedder a true/false is the best we can count on.

> I'd say feel free to refactor / change in a future PR (or block this one). I spent enough energy on it myself.

Sorry if this is untimely - feel free to land it, just trying to address the previous comments about release/debug differences.